src/lib/crypto/kimchi_backend/common/bigint.ml

```diff
@@ -109,5 +113,7 @@ module Make
       of_bytes bytes
   end)
 
-  let of_numeral s ~base = of_numeral s (String.length s) base
+  let of_numeral s ~base =
+    let s = Bytes.of_string s in
```

Comment:
This makes a copy, which we would like to avoid by giving the correct type above.

Reply:
any reason why of_numeral takes a `string` instead of a `bytes`?